lib/service/ApproveButton.dart

```diff
@@ -180,7 +180,7 @@ class _ApproveButtonBuyDialogState extends State<ApproveButtonBuyDialog> {
               showDialog(
                   context: context,
                   builder: (BuildContext context) =>
-                      widget.confirmDialog(context));
+                      widget.confirmDialog(context)).then((value) => Navigator.pop(context));
```

Comment:
This is unsafe since `context` shouldn't be used across async gaps.

You should await the `showDialog` and then do:
```
if (mounted) {
  Navigator.pop(context);
}
```